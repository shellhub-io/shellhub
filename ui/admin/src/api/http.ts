import { AxiosInstance } from "axios";
import { setupInterceptorsTo } from "./interceptors";
import * as axiosTs from "./client";
import { Configuration } from "./client";
import { BaseAPI } from "./client/base";

const configuration = new Configuration();
configuration.basePath = `${window.location.protocol}//${window.location.host}`;
configuration.accessToken = localStorage.getItem("cloud_token") || "";

// eslint-disable-next-line import/no-mutable-exports
let adminApi = new axiosTs.AdminApi(configuration);
// eslint-disable-next-line import/no-mutable-exports
let cloudApi = new axiosTs.CloudApi(configuration);

export const createNewClient = () => {
  const newConfiguration = new Configuration();
  newConfiguration.basePath = `${window.location.protocol}//${window.location.host}`;
  newConfiguration.accessToken = localStorage.getItem("cloud_token") || "";
  adminApi = new axiosTs.AdminApi(newConfiguration);
  cloudApi = new axiosTs.CloudApi(newConfiguration);
  return { adminApi, cloudApi };
};

declare module "./client/base" {
  interface BaseAPI {
    getAxios(): AxiosInstance;
    getConfiguration(): Configuration | undefined;
    setConfiguration(configuration: Configuration): void;
  }
}

/** Returns the axios instance */
BaseAPI.prototype.getAxios = function getAxios(this: BaseAPI): AxiosInstance {
  return this.axios;
};

/** Returns the configuration */
// eslint-disable-next-line vue/max-len
BaseAPI.prototype.getConfiguration = function getConfiguration(this: BaseAPI): Configuration | undefined {
  return this.configuration;
};

/** Sets the configuration */
// eslint-disable-next-line vue/max-len
BaseAPI.prototype.setConfiguration = function setConfiguration(this: BaseAPI, configuration: Configuration): void {
  this.configuration = configuration;
};

// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
setupInterceptorsTo(adminApi.axios);

export { adminApi, cloudApi };
