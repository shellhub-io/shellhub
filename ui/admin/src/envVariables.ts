/* eslint-disable import/prefer-default-export */
export const envVariables = {
  announcementsEnable: process.env.SHELLHUB_ANNOUNCEMENTS === "true",
  tinyMceKey: process.env.TINYMCE_API_KEY,
};
