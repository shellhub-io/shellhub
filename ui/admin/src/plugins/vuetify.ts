// Styles
// import "@mdi/font/css/materialdesignicons.css";
import "vuetify/styles";

// Vuetify
import { createVuetify } from "vuetify";

export default createVuetify({
  theme: {
    defaultTheme: "dark",
  },
});
