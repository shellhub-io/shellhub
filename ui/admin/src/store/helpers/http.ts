import Axios from "axios";
// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore //* use while route mapping is not in TypeScript
export default () => {
  const axios = Axios.create({
    baseURL: `${window.location.protocol}//${window.location.host}/admin`,
    headers: {
      Authorization: `Bearer ${localStorage.getItem("cloud_token")}`,
    },
  });

  return axios;
};
