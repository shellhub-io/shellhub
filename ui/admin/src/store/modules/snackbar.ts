import { Module } from "vuex";
import { State } from "./../index";

export interface snackbarMessageAndContentType {
  typeMessage: string;
  typeContent: string;
}

export interface SnackbarState {
  snackbarError: boolean;
  snackbarSuccess: boolean;
  snackbarMessageAndContentType: snackbarMessageAndContentType;
  snackbarCopy: boolean;
}

export const snackbar: Module<SnackbarState, State> = {
  namespaced: true,

  state: {
    snackbarError: false,
    snackbarSuccess: false,
    snackbarMessageAndContentType: { typeMessage: "", typeContent: "" },
    snackbarCopy: false,
  },

  getters: {
    snackbarSuccess: (state) => state.snackbarSuccess,
    snackbarError: (state) => state.snackbarError,
    snackbarMessageAndContentType: (state) => state.snackbarMessageAndContentType,
    snackbarCopy: (state) => state.snackbarCopy,
  },

  mutations: {
    setSnackbarSuccessAction: (state, data) => {
      state.snackbarMessageAndContentType = data;
      state.snackbarSuccess = true;
    },

    setSnackbarNoContent: (state, data) => {
      state.snackbarMessageAndContentType = data;
      state.snackbarSuccess = true;
    },

    setSnackbarSuccessDefault: (state) => {
      state.snackbarMessageAndContentType = { typeMessage: "default", typeContent: "" };
      state.snackbarSuccess = true;
    },

    unsetSnackbarSuccess: (state) => {
      state.snackbarSuccess = false;
    },

    setSnackbarErrorLoadingOrAction: (state, data) => {
      state.snackbarMessageAndContentType = data;
      state.snackbarError = true;
    },

    setSnackbarErrorCustom: (state, data) => {
      state.snackbarMessageAndContentType = data;
      state.snackbarError = true;
    },

    setSnackbarErrorDefault: (state) => {
      state.snackbarMessageAndContentType = { typeMessage: "default", typeContent: "" };
      state.snackbarError = true;
    },

    setSnackbarErrorLicense: (state, data) => {
      state.snackbarMessageAndContentType = data;
      state.snackbarError = true;
    },

    unsetSnackbarError: (state) => {
      state.snackbarError = false;
    },

    setSnackbarCopy: (state, value) => {
      state.snackbarMessageAndContentType = { typeMessage: "", typeContent: value };
      state.snackbarCopy = true;
    },

    unsetSnackbarCopy: (state) => {
      state.snackbarCopy = false;
    },
  },

  actions: {
    showSnackbarSuccessAction: ({ commit }, value) => {
      const data = { typeMessage: "action", typeContent: value };
      commit("setSnackbarSuccessAction", data);
    },

    showSnackbarSuccessDefault: ({ commit }) => {
      commit("setSnackbarSuccessDefault");
    },

    unsetShowStatusSnackbarSuccess: ({ commit }) => {
      commit("unsetSnackbarSuccess");
    },

    showSnackbarErrorLoading: ({ commit }, value) => {
      const data = { typeMessage: "loading", typeContent: value };
      commit("setSnackbarErrorLoadingOrAction", data);
    },

    showSnackbarNoContent: ({ commit }) => {
      const data = { typeMessage: "no-content", typeContent: "" };
      commit("setSnackbarNoContent", data);
    },

    showSnackbarErrorAction: ({ commit }, value) => {
      const data = { typeMessage: "action", typeContent: value };
      commit("setSnackbarErrorLoadingOrAction", data);
    },

    showSnackbarErrorCustom: ({ commit }, value) => {
      const data = { typeMessage: "custom", typeContent: value };
      commit("setSnackbarErrorCustom", data);
    },

    showSnackbarErrorLicense: ({ commit }, value) => {
      const data = { typeMessage: "licenseRequired", typeContent: value };
      commit("setSnackbarErrorLicense", data);
    },

    showSnackbarErrorDefault: ({ commit }) => {
      commit("setSnackbarErrorDefault");
    },

    unsetShowStatusSnackbarError: ({ commit }) => {
      commit("unsetSnackbarError");
    },

    showSnackbarCopy: ({ commit }, value) => {
      commit("setSnackbarCopy", value);
    },

    unsetShowStatusSnackbarCopy: ({ commit }) => {
      commit("unsetSnackbarCopy");
    },
  },
};
