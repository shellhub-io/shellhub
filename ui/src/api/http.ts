import axios, { AxiosInstance } from "axios";
import * as axiosTs from "./client";
import { Configuration } from "./client";
import { BaseAPI } from "./client/base";
import { setupInterceptorsTo } from "./interceptors";

// This is the default configuration for local instance endpoints
const configuration = new Configuration();
configuration.basePath = `${window.location.protocol}//${window.location.host}`;
configuration.accessToken = localStorage.getItem("token") || "";

// We need a custom configuration for cloud endpoints
const cloudApiConfiguration = new Configuration();
cloudApiConfiguration.basePath = "https://cloud.shellhub.io";

// Creates a new axios instance and setup interceptors by default
const newAxiosInstance = (setupInterceptor = true): AxiosInstance => {
  const instance = axios.create();
  if (setupInterceptor) setupInterceptorsTo(instance);
  return instance;
};
const sessionsApi = new axiosTs.SessionsApi(
  configuration,
  undefined,
  newAxiosInstance(),
);
const devicesApi = new axiosTs.DevicesApi(
  configuration,
  undefined,
  newAxiosInstance(),
);
const containersApi = new axiosTs.ContainersApi(
  configuration,
  undefined,
  newAxiosInstance(),
);
const systemApi = new axiosTs.SystemApi(
  configuration,
  undefined,
  newAxiosInstance(),
);
const namespacesApi = new axiosTs.NamespacesApi(
  configuration,
  undefined,
  newAxiosInstance(),
);
const apiKeysApi = new axiosTs.ApiKeysApi(
  configuration,
  undefined,
  newAxiosInstance(),
);
const sshApi = new axiosTs.SshApi(
  configuration,
  undefined,
  newAxiosInstance(),
);
const tagsApi = new axiosTs.TagsApi(
  configuration,
  undefined,
  newAxiosInstance(),
);
const usersApi = new axiosTs.UsersApi(
  configuration,
  undefined,
  newAxiosInstance(),
);
const mfaApi = new axiosTs.MfaApi(
  configuration,
  undefined,
  newAxiosInstance(),
);
const tunnelApi = new axiosTs.TunnelsApi(
  configuration,
  undefined,
  newAxiosInstance(),
);
const billingApi = new axiosTs.BillingApi(
  configuration,
  undefined,
  newAxiosInstance(),
);
const rulesApi = new axiosTs.RulesApi(
  configuration,
  undefined,
  newAxiosInstance(),
);
const announcementApi = new axiosTs.AnnouncementsApi(
  cloudApiConfiguration,
  undefined,
  newAxiosInstance(false),
);

/**
 * Extends the interface BaseAPI to include a new method `getAxios` allowing
 * access to the protected axios property without the need to access it
 * directly from outside the class and avoiding a linting error caused by
 * accessing a protected property.
 **/
declare module "./client/base" {
  interface BaseAPI {
    getAxios(): AxiosInstance;
    getConfiguration(): Configuration | undefined;
    setConfiguration(configuration: Configuration): void;
  }
}

/** Returns the axios instance */
BaseAPI.prototype.getAxios = function getAxios(this: BaseAPI): AxiosInstance {
  return this.axios;
};

/** Returns the configuration */
BaseAPI.prototype.getConfiguration = function getConfiguration(this: BaseAPI): Configuration | undefined {
  return this.configuration;
};

/** Sets the configuration */
BaseAPI.prototype.setConfiguration = function setConfiguration(this: BaseAPI, configuration: Configuration): void {
  this.configuration = configuration;
};

// Reloads the configuration for all APIs
const reloadConfiguration = () => {
  [
    sessionsApi,
    devicesApi,
    containersApi,
    systemApi,
    namespacesApi,
    apiKeysApi,
    sshApi,
    tagsApi,
    usersApi,
    mfaApi,
    tunnelApi,
    billingApi,
    rulesApi,
  ].forEach((api) => {
    api.setConfiguration(configuration);
  });
};

export {
  configuration,
  reloadConfiguration,
  sessionsApi,
  devicesApi,
  containersApi,
  systemApi,
  namespacesApi,
  apiKeysApi,
  sshApi,
  tagsApi,
  usersApi,
  mfaApi,
  tunnelApi,
  billingApi,
  rulesApi,
  announcementApi,
};
