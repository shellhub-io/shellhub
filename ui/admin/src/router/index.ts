import { createRouter, createWebHistory, NavigationGuardNext, RouteLocationNormalized } from "vue-router";
import Login from "@admin/views/Login.vue";
import Dashboard from "@admin/views/Dashboard.vue";
import Users from "@admin/views/Users.vue";
import SettingsLicense from "@admin/components/Settings/SettingsLicense.vue";
import SettingsAuthentication from "@admin/components/Settings/SettingsAuthentication.vue";
import Namespaces from "@admin/views/Namespaces.vue";
import Settings from "@admin/views/Settings.vue";
import { computed } from "vue";
import useLicenseStore from "@admin/store/modules/license";
import useLayoutStore from "@admin/store/modules/layout";
import useAuthStore from "@admin/store/modules/auth";
import { plugin as snackbar } from "@/plugins/snackbar"; // using direct plugin because inject() doesn't work outside components

const routes = [
  {
    path: "/login",
    name: "login",
    component: Login,
    meta: {
      layout: "SimpleLayout",
      requiresAuth: false,
    },
  },
  {
    path: "/",
    name: "dashboard",
    component: Dashboard,
  },
  {
    path: "/users",
    name: "users",
    component: Users,
  },
  {
    path: "/user/:id",
    name: "userDetails",
    component: () => import("@admin/views/UserDetails.vue"),
  },
  {
    path: "/devices",
    name: "devices",
    component: () => import("@admin/views/Device.vue"),
    redirect: {
      name: "listDevices",
    },
    children: [
      {
        path: "",
        name: "listDevices",
        component: () => import("@admin/views/Device.vue"),
      },
    ],
  },
  {
    path: "/device/:id",
    name: "deviceDetails",
    component: () => import("@admin/views/DeviceDetails.vue"),
  },
  {
    path: "/sessions",
    name: "sessions",
    component: () => import("@admin/views/Sessions.vue"),
  },
  {
    path: "/session/:id",
    name: "sessionDetails",
    component: () => import("@admin/views/SessionDetails.vue"),
  },
  {
    path: "/settings",
    name: "Settings",
    component: Settings,
    redirect: { name: "SettingProfile" },
    children: [
      {
        path: "authentication",
        name: "SettingAuthentication",
        component: SettingsAuthentication,
      },
      {
        path: "license",
        name: "SettingLicense",
        component: SettingsLicense,
      },
    ],
  },
  {
    path: "/firewall-rules",
    name: "firewall-rules",
    component: () => import("@admin/views/FirewallRules.vue"),
  },
  {
    path: "/firewall-rules/:id",
    name: "firewallRulesDetails",
    component: () => import("@admin/views/FirewallRulesDetails.vue"),
  },
  {
    path: "/namespaces",
    name: "namespaces",
    component: Namespaces,
  },
  {
    path: "/namespace/:id",
    name: "namespaceDetails",
    component: () => import("@admin/views/NamespaceDetails.vue"),
  },
  {
    path: "/announcements",
    name: "announcements",
    component: () => import("@admin/views/Announcements.vue"),
  },
  {
    path: "/announcement/:uuid",
    name: "announcementDetails",
    component: () => import("@admin/views/AnnouncementDetails.vue"),
  },
  {
    path: "/new-announcement",
    name: "new-announcement",
    component: () => import("@admin/views/NewAnnouncement.vue"),
  },
];

const router = createRouter({
  history: createWebHistory("/admin/"),
  routes,
});

router.beforeEach(
  async (to: RouteLocationNormalized, from: RouteLocationNormalized, next: NavigationGuardNext) => {
    const licenseStore = useLicenseStore();
    const layoutStore = useLayoutStore();
    const authStore = useAuthStore();

    const requiresAuth = to.meta.requiresAuth ?? true;

    const layout = to.meta.layout || "AppLayout";
    layoutStore.setLayout(layout as string);

    if (!authStore.isLoggedIn && requiresAuth) {
      return next({
        name: "login",
        query: { redirect: to.fullPath },
      });
    }

    if (authStore.isLoggedIn && !to.meta.requiresAuth) {
      const license = computed(() => licenseStore.getLicense);

      try {
        await licenseStore.get();

        if (license.value.expired && to.name !== "SettingLicense") {
          snackbar.showError("Your license has expired. Please update it and try again.");
          return next({ name: "SettingLicense" });
        }
      } catch {
        if (to.name !== "SettingLicense") {
          snackbar.showError("Failed to get your license info. Please check it and try again.");
          return next({ name: "SettingLicense" });
        }
      }
    }

    return next();
  },
);

export default router;
