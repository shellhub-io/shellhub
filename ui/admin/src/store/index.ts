import { InjectionKey } from "vue";
import { createStore, Store, useStore as vuexUseStore } from "vuex";
import { stats, StatsState } from "./modules/stats";

import { UsersState, users } from "./modules/users";
import { AuthState, auth } from "./modules/auth";
import { DevicesState, devices } from "./modules/devices";
import { firewallRules, FirewallRulesState } from "./modules/firewall_rules";
import { namespaces, NamespacesState } from "./modules/namespaces";
import { layout, LayoutState } from "./modules/layout";
import { license, LicenseState } from "./modules/license";
import { publicKeys, PublicKeysState } from "./modules/public_keys";
import { sessions, SessionsState } from "./modules/sessions";
import { snackbar, SnackbarState } from "./modules/snackbar";
import { spinner, SpinnerState } from "./modules/spinner";
import { announcement, AnnouncementState } from "./modules/announcement";
import { instance, InstanceState } from "./modules/instance";

export interface State {
  user: UsersState;
  auth: AuthState;
  devices: DevicesState;
  firewallRules: FirewallRulesState;
  layout: LayoutState;
  license: LicenseState;
  namespaces: NamespacesState;
  publicKeys: PublicKeysState;
  sessions: SessionsState;
  snackbar: SnackbarState;
  spinner: SpinnerState;
  stats: StatsState;
  announcement: AnnouncementState;
  instance: InstanceState;
}

// eslint-disable-next-line symbol-description
export const key: InjectionKey<Store<State>> = Symbol();

export const store = createStore<State>({
  modules: {
    users,
    auth,
    devices,
    firewallRules,
    layout,
    license,
    namespaces,
    publicKeys,
    sessions,
    snackbar,
    spinner,
    stats,
    announcement,
    instance,
  },
});

export function useStore(): Store<State> {
  return vuexUseStore(key);
}
