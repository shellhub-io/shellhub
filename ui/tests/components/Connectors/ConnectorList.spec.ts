import { mount, VueWrapper } from "@vue/test-utils";
import { createVuetify } from "vuetify";
import MockAdapter from "axios-mock-adapter";
import { expect, describe, it, beforeEach, vi } from "vitest";
import { store, key } from "@/store";
import ConnectorList from "@/components/Connector/ConnectorList.vue";
import { envVariables } from "@/envVariables";
import { router } from "@/router";
import { namespacesApi, billingApi, devicesApi } from "@/api/http";
import { SnackbarPlugin } from "@/plugins/snackbar";

type ConnectorListWrapper = VueWrapper<InstanceType<typeof ConnectorList>>;

describe("Connector List", () => {
  let wrapper: ConnectorListWrapper;

  const vuetify = createVuetify();

  let mockNamespace: MockAdapter;

  let mockDevices: MockAdapter;

  let mockBilling: MockAdapter;

  const devices = [
    {
      uid: "a582b47a42d",
      name: "39-5e-2a",
      identity: {
        mac: "00:00:00:00:00:00",
      },
      info: {
        id: "linuxmint",
        pretty_name: "Linux Mint 19.3",
        version: "",
      },
      public_key: "----- PUBLIC KEY -----",
      tenant_id: "fake-tenant-data",
      last_seen: "2020-05-20T18:58:53.276Z",
      online: false,
      namespace: "user",
      status: "accepted",
      tags: ["test"],
    },
    {
      uid: "a582b47a42e",
      name: "39-5e-2b",
      identity: {
        mac: "00:00:00:00:00:00",
      },
      info: {
        id: "linuxmint",
        pretty_name: "Linux Mint 19.3",
        version: "",
      },
      public_key: "----- PUBLIC KEY -----",
      tenant_id: "fake-tenant-data",
      last_seen: "2020-05-20T19:58:53.276Z",
      online: true,
      namespace: "user",
      status: "accepted",
      tags: ["test"],
    },
  ];

  const members = [
    {
      id: "xxxxxxxx",
      username: "test",
      role: "owner",
    },
  ];

  const connectors = {
    data: [
      {

        uid: "3dd0d1f8-8246-4519-b11a-a3dd33717f65",
        tenant_id: "3dd0d1f8-8246-4519-b11a-a3dd33717f65",
        enable: true,
        address: "127.0.0.1",
        port: 2375,
        secure: false,
        status:

        {
          State: "connected",
          Message: "",
        },
        tls: null,

      },
    ],
    headers: {
      "x-total-count": 1,
    },
  };

  const billingData = {
    active: false,
    status: "canceled",
    customer_id: "cus_test",
    subscription_id: "sub_test",
    current_period_end: 2068385820,
    created_at: "",
    updated_at: "",
    invoices: [],
  };

  const namespaceData = {
    name: "user",
    owner: "xxxxxxxx",
    tenant_id: "fake-tenant-data",
    members,
    max_devices: 3,
    devices_count: 3,
    devices: 2,
    created_at: "",
    billing: billingData,
  };

  const authData = {
    status: "",
    token: "",
    user: "test",
    name: "test",
    tenant: "fake-tenant-data",
    email: "test@test.com",
    id: "xxxxxxxx",
    role: "owner",
  };

  const customerData = {
    id: "cus_test",
    name: "test",
    email: "test@test.com",
    payment_methods: [
      {
        id: "test_id",
        number: "xxxxxxxxxxxx4242",
        brand: "visa",
        exp_month: 3,
        exp_year: 2029,
        cvc: "",
        default: true,
      },
    ],
  };

  const stats = {
    registered_devices: 2,
    online_devices: 1,
    active_sessions: 0,
    pending_devices: 0,
    rejected_devices: 0,
  };

  beforeEach(async () => {
    window.matchMedia = vi.fn().mockImplementation((query) => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: vi.fn(),
      removeListener: vi.fn(),
      addEventListener: vi.fn(),
      removeEventListener: vi.fn(),
      dispatchEvent: vi.fn(),
    }));

    vi.useFakeTimers();
    localStorage.setItem("tenant", "fake-tenant-data");
    envVariables.isCloud = true;

    mockBilling = new MockAdapter(billingApi.getAxios());
    mockNamespace = new MockAdapter(namespacesApi.getAxios());
    mockDevices = new MockAdapter(devicesApi.getAxios());

    mockNamespace.onGet("http://localhost:3000/api/namespaces/fake-tenant-data").reply(200, namespaceData);
    mockBilling.onGet("http://localhost:3000/api/billing/customer").reply(200, customerData);
    mockBilling.onGet("http://localhost:3000/api/billing/subscription").reply(200, billingData);
    mockBilling.onGet("http://localhost:3000/api/billing/devices-most-used").reply(200, devices);
    mockNamespace.onGet("http://localhost:3000/api/connector?page=1&per_page=10").reply(200, connectors);
    mockDevices.onGet("http://localhost:3000/api/stats").reply(200, stats);

    store.commit("auth/authSuccess", authData);
    store.commit("namespaces/setNamespace", namespaceData);
    store.commit("billing/setSubscription", billingData);
    store.commit("customer/setCustomer", customerData);
    store.commit("connectors/setConnectors", connectors);

    wrapper = mount(ConnectorList, {
      global: {
        plugins: [[store, key], vuetify, router, SnackbarPlugin],
      },
    });
  });

  it("Is a Vue instance", () => {
    expect(wrapper.vm).toBeTruthy();
  });

  it("Renders the component", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("Renders the component HTML", async () => {
    expect(wrapper.findComponent('[data-test="connector-list"]').exists()).toBe(true);
    expect(wrapper.find('[data-test="status-connector"]').exists()).toBe(true);
    expect(wrapper.findComponent('[data-test="switch-enable"]').exists()).toBe(true);
    expect(wrapper.findComponent('[data-test="ip-chip"]').exists()).toBe(true);
    expect(wrapper.findComponent('[data-test="secure-icon"]').exists()).toBe(true);
    expect(wrapper.find('[data-test="menu-key-component"]').exists()).toBe(true);
    expect(wrapper.findComponent('[data-test="connector-list-actions"]').exists()).toBe(true);
    expect(wrapper.findComponent('[data-test="mdi-information-list-item"]').exists()).toBe(true);
    expect(wrapper.findComponent('[data-test="no-connector-validate"]').exists()).toBe(false);
  });
});
