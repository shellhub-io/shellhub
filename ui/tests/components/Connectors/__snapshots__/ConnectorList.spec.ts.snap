// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Connector List > Renders the component 1`] = `
"<div data-v-046d4556="" data-v-6fdd6bd6="" data-test="connector-list">
  <div data-v-046d4556="" class="v-table v-theme--light v-table--density-default bg-background border rounded">
    <!---->
    <div class="v-table__wrapper">
      <table>
        <thead data-v-046d4556="" class="bg-v-theme-background">
          <tr data-v-046d4556="">
            <th data-v-046d4556="" class="text-center"><span data-v-046d4556="">Status</span></th>
            <th data-v-046d4556="" class="text-center"><span data-v-046d4556="">Enable</span></th>
            <th data-v-046d4556="" class="text-center"><span data-v-046d4556="">Host</span></th>
            <th data-v-046d4556="" class="text-center"><span data-v-046d4556="">Secure</span></th>
            <th data-v-046d4556="" class="text-center"><span data-v-046d4556="">Actions</span></th>
          </tr>
        </thead>
        <tbody data-v-046d4556="">
          <tr data-v-6fdd6bd6="">
            <td data-v-6fdd6bd6="" class="text-center">
              <div data-v-6fdd6bd6="" data-test="status-connector" class="disabled text-center"></div>
            </td>
            <td data-v-6fdd6bd6="" class="text-center">
              <div data-v-6fdd6bd6="" class="v-row">
                <div data-v-6fdd6bd6="" class="v-col d-flex justify-center" data-test="switch-enable">
                  <div data-v-6fdd6bd6="" class="v-input v-input--horizontal v-input--center-affix v-input--density-default v-theme--light v-locale--is-ltr v-input--dirty v-switch v-switch--inset">
                    <!---->
                    <div class="v-input__control">
                      <div class="v-selection-control v-selection-control--dirty v-selection-control--density-default">
                        <div class="v-selection-control__wrapper text-primary">
                          <div class="v-switch__track bg-primary">
                            <!---->
                            <!---->
                          </div>
                          <div class="v-selection-control__input"><input checked="" id="switch-v-0" aria-disabled="false" type="checkbox" aria-describedby="switch-v-0-messages" value="true">
                            <div class="v-switch__thumb">
                              <transition-stub name="scale-transition" appear="false" persisted="false" css="true">
                                <!---->
                              </transition-stub>
                            </div>
                          </div>
                        </div>
                        <!---->
                      </div>
                    </div>
                    <!---->
                    <!---->
                  </div>
                </div>
              </div>
            </td>
            <td data-v-6fdd6bd6="" class="text-center"><span data-v-6fdd6bd6="" class="v-chip v-theme--light v-chip--density-default v-chip--size-default v-chip--variant-tonal" draggable="false" data-test="ip-chip"><!----><span class="v-chip__underlay"></span>
              <!---->
              <!---->
              <div class="v-chip__content" data-no-activator=""><span data-v-6fdd6bd6="" aria-describedby="v-tooltip-v-4" class="hover-text">127.0.0.1:2375</span>
                <!--teleport start-->
                <!--teleport end-->
              </div>
              <!---->
              <!----></span>
            </td>
            <td data-v-6fdd6bd6="" class="text-center"><i data-v-6fdd6bd6="" class="mdi-lock-open-alert mdi v-icon notranslate v-theme--light text-grey-darken-2 mr-1" style="font-size: 26px; height: 26px; width: 26px;" aria-hidden="true" data-test="secure-icon"></i></td>
            <td data-v-6fdd6bd6="" class="text-center" data-test="menu-key-component"><button data-v-6fdd6bd6="" type="button" class="v-btn v-btn--icon v-theme--light v-btn--density-comfortable v-btn--size-default v-btn--variant-plain border rounded bg-v-theme-background" aria-haspopup="menu" aria-expanded="false" aria-controls="v-menu-v-5" data-test="connector-list-actions"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
                <!----><span class="v-btn__content" data-no-activator=""><i class="mdi-format-list-bulleted mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i></span>
                <!---->
                <!---->
              </button>
              <!--teleport start-->
              <!--teleport end-->
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <!---->
  </div>
  <div data-v-046d4556="" class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 4px; --v-progress-linear-height: 4px;" role="progressbar" aria-hidden="false" aria-valuemin="0" aria-valuemax="100" alt="Data table loading">
    <!---->
    <div class="v-progress-linear__background"></div>
    <div class="v-progress-linear__buffer" style="width: 0%;"></div>
    <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
      <div class="v-progress-linear__indeterminate">
        <div class="v-progress-linear__indeterminate long"></div>
        <div class="v-progress-linear__indeterminate short"></div>
      </div>
    </transition-stub>
    <!---->
  </div>
  <div data-v-046d4556="" class="d-flex w-100 justify-end align-center"><span data-v-046d4556="" class="text-subtitle-2 mr-4">Items per page:</span>
    <div data-v-046d4556="">
      <div data-v-046d4556="" class="v-input v-input--horizontal v-input--density-default v-theme--light v-locale--is-ltr v-input--dirty v-text-field v-input--plain-underlined v-combobox v-combobox--single mb-4">
        <!---->
        <div class="v-input__control">
          <div class="v-field v-field--active v-field--appended v-field--dirty v-field--no-label v-field--variant-underlined v-theme--light v-locale--is-ltr" role="combobox" aria-haspopup="menu" aria-expanded="false" aria-controls="v-menu-v-11">
            <div class="v-field__overlay"></div>
            <div class="v-field__loader">
              <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
                <!---->
                <div class="v-progress-linear__background"></div>
                <div class="v-progress-linear__buffer" style="width: 0%;"></div>
                <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
                  <div class="v-progress-linear__indeterminate">
                    <div class="v-progress-linear__indeterminate long"></div>
                    <div class="v-progress-linear__indeterminate short"></div>
                  </div>
                </transition-stub>
                <!---->
              </div>
            </div>
            <!---->
            <div class="v-field__field" data-no-activator="">
              <!---->
              <!---->
              <!---->
              <div class="v-field__input" data-no-activator="">
                <!---->
                <!---->
                <div class="v-combobox__selection"><span class="v-combobox__selection-text">10<!----></span></div><input size="1" type="text" id="input-v-9" aria-describedby="input-v-9-messages" outlined="" value="10">
              </div>
              <!---->
            </div>
            <!---->
            <div class="v-field__append-inner">
              <!----><i class="mdi-menu-down mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable v-combobox__menu-icon" role="button" aria-hidden="false" tabindex="-1" aria-label="Open" title="Open"></i>
              <!---->
            </div>
            <div class="v-field__outline">
              <!---->
              <!---->
            </div>
          </div>
        </div>
        <!---->
        <!---->
      </div>
    </div>
    <div data-v-046d4556="" class="d-flex align-center"><button data-v-046d4556="" type="button" class="v-btn v-btn--disabled v-btn--icon v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-plain" disabled=""><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
        <!----><span class="v-btn__content" data-no-activator=""><i class="mdi-chevron-left mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i></span>
        <!---->
        <!---->
      </button><span data-v-046d4556="" class="text-subtitle-2">1 of 1</span><button data-v-046d4556="" type="button" class="v-btn v-btn--disabled v-btn--icon v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-plain" disabled=""><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
        <!----><span class="v-btn__content" data-no-activator=""><i class="mdi-chevron-right mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i></span>
        <!---->
        <!---->
      </button></div>
  </div>
</div>"
`;
